tasks/managed/publish-to-mrrc/publish-to-mrrc.yaml

```diff
@@ -42,7 +42,16 @@ spec:
           echo "Downloading the maven repo zip $r"
           SOURCE_REPO=${r%%@sha256:*}
           AUTH_FILE=$(mktemp)
-          select-oci-auth "${SOURCE_REPO}" > "$AUTH_FILE"
+          registry=$(echo "${SOURCE_REPO}" | cut -d '/' -f 1)
+          if [ "$registry" = "docker.io" ]; then
+            # For docker.io, the auth key will always be https://index.docker.io/v1/
+            select-oci-auth "${SOURCE_REPO}" > "$AUTH_FILE"
+          else
+            # For other registries, the auth key will be modified to the full repository path, so that
+            # we can create a combined auth file with source and destination entries for `cosign copy` later
+            select-oci-auth "${SOURCE_REPO}" | jq -c \
+              '.auths."'"${SOURCE_REPO}"'" = .auths."'"$registry"'" | del(.auths."'"$registry"'")' > "$AUTH_FILE"
+          fi
```

Comment:
I don't think you need any of this. The special logic is needed in push-snapshot, because we need to contain both the source and target repo credentials in one docker config file - we copy the image from one place to another. Here you're only pulling, so a simple `select-oci-auth` call should be enough.

What you should do instead is update the image used in this task (and update the tests to use the same). The one currently used is from August 2024, so it definitely does not contain support for docker.io.

(Note that we plan to enable automatic updates of the utils image references in this repo via Renovate/Mintmaker quite soon.)

Reply:
Ok. I've updated in the new commit. Thanks a lot for the info!

Reply:
Sure of course. Just updated.